pkg/postage/service.go

```diff
@@ -116,29 +116,29 @@ func (ps *service) HandleCreate(b *Batch) error {
 
 // HandleTopUp implements the BatchEventListener interface. This is fired on receiving
 // a batch topup event from the blockchain to update stampissuer details
-func (ps *service) HandleTopUp(batchID []byte, newValue *big.Int) {
+func (ps *service) HandleTopUp(batchID []byte, amount *big.Int) {
 	ps.lock.Lock()
 	defer ps.lock.Unlock()
 
 	for _, v := range ps.issuers {
 		if bytes.Equal(batchID, v.data.BatchID) {
-			if newValue.Cmp(v.data.BatchAmount) > 0 {
-				v.data.BatchAmount = newValue
+			if amount.Cmp(v.data.BatchAmount) > 0 {
+				v.data.BatchAmount.Add(v.data.BatchAmount, amount)
 			}
 			return
 		}
 	}
 }
 
-func (ps *service) HandleDepthIncrease(batchID []byte, newDepth uint8, normalisedBalance *big.Int) {
+func (ps *service) HandleDepthIncrease(batchID []byte, newDepth uint8, newBatchAmount *big.Int) {
 	ps.lock.Lock()
 	defer ps.lock.Unlock()
 
 	for _, v := range ps.issuers {
 		if bytes.Equal(batchID, v.data.BatchID) {
 			if newDepth > v.data.BatchDepth {
 				v.data.BatchDepth = newDepth
-				v.data.BatchAmount = normalisedBalance
+				v.data.BatchAmount = newBatchAmount
```

Comment:
`HandleDepthIncrease` gets passed the balance, not the amount. You cannot update the amount with balance. 
You can remove the balance argument.